privateSBOMExchange/src/cpabe/src/lib.rs

```diff
@@ -0,0 +1,113 @@
+use pyo3::prelude::*;
+use pyo3::types::PyList;
+use rabe::schemes::bsw::*;
+use rabe::utils::policy::pest::PolicyLanguage;
+use serde_json::{ser, de};
+use std::vec::Vec;
+//use serde_json::{Result};
+
+/// Formats the sum of two numbers as string.
+#[pyfunction]
+fn sum_as_string(a: usize, b: usize) -> PyResult<String> {
+    Ok((a + b).to_string())
+}
+
+#[pyfunction]
+fn cpabe_setup() -> (String, String) {
+    let (pk, mk) = setup();
+    let spk = pk_to_json(pk);
+    let smk = mk_to_json(mk);
+    (spk, smk)
+}
+
+#[pyfunction]
+fn cpabe_keygen<'py>(pk: String, mk: String, attributes: &Bound<'py, PyList>) -> String {
+    let _mk = json_to_mk(mk.clone());
+    let _pk = json_to_pk(pk.clone());
+    let mut attr_vec: Vec<String> = Vec::new();
+    for value in attributes.iter() {
+        attr_vec.push(value.to_string());
+    }
+    
+    let pos: Vec<&str> = attr_vec.iter().map(AsRef::as_ref).collect();
+    let sk = keygen(&_pk, &_mk, pos.as_slice()).unwrap();
+    sk_to_json(sk)
+}
+
+
+#[pyfunction]
+fn cpabe_encrypt<'py>(pk: String, policy: &str, plaintext: &[u8]) -> String {
+    let _pk = json_to_pk(pk.clone());
+    let ct = encrypt(&_pk, policy, PolicyLanguage::HumanPolicy, plaintext).unwrap();
+    ct_to_json(ct)
+}
+
+#[pyfunction]
+fn cpabe_decrypt(sk: String, ct: String) -> Vec<u8> {
+    let _sk = json_to_sk(sk.clone());
+    let _ct = json_to_ct(ct.clone());
+
+    let pt = decrypt(&_sk, &_ct).unwrap();
+    pt
+
+}
+
+#[pyfunction]
+fn cpabe_delegate<'py>(pk: String, sk: String, subset: &Bound<'py, PyList>) -> String {
+
+    let _sk = json_to_sk(sk.clone());
+    let _pk = json_to_pk(pk.clone());
+    let mut attr_vec: Vec<String> = Vec::new();
+    for value in subset.iter() {
+        attr_vec.push(value.to_string());
+    }
+    
+    let pos: Vec<&str> = attr_vec.iter().map(AsRef::as_ref).collect();
+    let sk = delegate(&_pk, &_sk, pos.as_slice()).unwrap();
+    sk_to_json(sk)
+}
+
+
+fn pk_to_json(a: CpAbePublicKey) -> String {
```

Comment:
I'm assuming this means keys won't be in PEM format? We might want to document this.

Reply:
yeah, I don't think that cp-abe keys have a supported PEM format without shoehorning them in another format. I'll add comments in a new commit

Reply:
Thanks! Yeah, those comments will be a helpful note to self that you can't just import keys etc.